service/history/transferQueueProcessor.go

```diff
@@ -328,26 +328,26 @@ func (t *transferQueueProcessorImpl) processActivityTask(task *persistence.Trans
 	if err != nil {
 		return err
 	}
+	defer release()
 
-	var mb *mutableStateBuilder
-	mb, err = context.loadWorkflowExecution()
+	var msBuilder *mutableStateBuilder
+	msBuilder, err = context.loadWorkflowExecution()
 	timeout := int32(0)
 	if err != nil {
-		release()
 		if _, ok := err.(*workflow.EntityNotExistsError); ok {
 			// this could happen if this is a duplicate processing of the task, and the execution has already completed.
 			return nil
 		}
 		return err
 	}
 
-	if ai, found := mb.GetActivityInfo(task.ScheduleID); found {
+	if ai, found := msBuilder.GetActivityInfo(task.ScheduleID); found {
 		timeout = ai.ScheduleToStartTimeout
 	} else {
 		logging.LogDuplicateTransferTaskEvent(t.logger, persistence.TransferTaskTypeActivityTask, task.TaskID, task.ScheduleID)
 	}
-	release()
 
+	release()
```

Comment:
Can you add a comment on why calling the release here is important.

Reply:
sure